crossbeam-channel/src/channel.rs

```diff
@@ -1153,14 +1191,57 @@ impl<T> Receiver<T> {
             _ => false,
         }
     }
+
+    /// Disconnects the channel.
+    ///
+    /// Explicitly disconnects the channel and returns `true`, as if all instances of either side
+    /// (receiver and sender) are dropped, unless it has been already disconnected. Otherwise, this
+    /// method does nothing and returns `false`.  Also, this method does nothing and returns
+    /// `false` as well, if this method is called on channels other than bounded-capacity,
+    /// unbounded-capacity or zero-capacity channels.
+    ///
+    /// The successful disconnect operation results in immediately rejecting receiving messages,
+    /// unless they are sent already but not received yet.
+    ///
+    /// Disconnected channels can be restored connected by calling [`connect`], as long as there
+    /// are at least a sender and a receiver for the channel.
+    ///
```

Comment:
maybe mention about race condition among other disconnect()s and drop of the last instance of sender/receiver...